op-challenger/game/fault/contracts/oracle.go

```diff
@@ -43,3 +87,70 @@ func (c *PreimageOracleContract) AddGlobalDataTx(data *types.PreimageOracleData)
 	call := c.contract.Call(methodLoadKeccak256PreimagePart, new(big.Int).SetUint64(uint64(data.OracleOffset)), data.GetPreimageWithoutSize())
 	return call.ToTxCandidate()
 }
+
+func (c *PreimageOracleContract) InitLargePreimage(uuid *big.Int, partOffset uint32, claimedSize uint32) (txmgr.TxCandidate, error) {
+	call := c.contract.Call(methodInitLPP, uuid, partOffset, claimedSize)
+	return call.ToTxCandidate()
+}
+
+func (c *PreimageOracleContract) AddLeaves(uuid *big.Int, leaves []Leaf, finalize bool) ([]txmgr.TxCandidate, error) {
+	var txs []txmgr.TxCandidate
+	for _, leaf := range leaves {
+		commitments := [][32]byte{([32]byte)(leaf.StateCommitment.Bytes())}
+		call := c.contract.Call(methodAddLeavesLPP, uuid, leaf.Input[:], commitments, finalize)
```

Comment:
We can send more than one leaf in each call and should take advantage of that to reduce gas costs.  I'd suggest this method just sends the leaves its given in a single tx and we handle splitting into separate transactions in `large.go`.

Reply:
That makes sense, planning to clean this up quite a bit before moving out of draft 😄